trogging/src/lib.rs

```diff
@@ -444,11 +445,20 @@ where
 struct LimitedWriter<W: Write>(usize, W);
 impl<W: Write> Write for LimitedWriter<W> {
     fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        let truncated = &buf[..min(self.0, buf.len())];
+        if truncated[truncated.len() - 1] != b'\n' {
+            // slow path; copy buffer and append a newline at the end
+            // we still want to perform a single write syscall (if possible).
+            let mut tmp = Vec::new();
+            tmp.extend_from_slice(truncated);
```

Comment:
```suggestion
            let mut tmp = truncated.to_vec();
```

Reply:
Or:

```suggestion
            let mut tmp = Vec::with_capacity(truncated.len() + 1);
            tmp.extend_from_slice(truncated);
```

First one is shorted, second one might avoid that the data is copied again when the subsequent `push` hits the capacity limit.